staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -229,29 +235,32 @@ func (b *builder) descriptionFor(path, verb string) string {
 //     action can be one of: GET, PUT, PATCH, POST, DELETE;
 //     verb can be one of: list, read, replace, patch, create, delete, deletecollection;
 //     sample is the sample Go type for response type.
-func (b *builder) buildRoute(root, path, action, verb string, sample interface{}) *restful.RouteBuilder {
+func (b *builder) buildRoute(root, path, httpVerb, actionVerb, operationVerb string, sample interface{}, params []*restful.Parameter) *restful.RouteBuilder {
```

Comment:
s/httpVerb/httpMethod/

the difference between `actionVerb` and `operationVerb` is very difficult to understand. is one supposed to map to a verb used for authorization, and the other to an english verb used for documentation?

Reply:
godoc needs updating with final names and clearer explanations

Reply:
> the difference between actionVerb and operationVerb is very difficult to understand. is one supposed to map to a verb used for authorization, and the other to an english verb used for documentation?

"actionVerb" is for marking in the vendored-extension, and it's also in consistent w/ authorization. 

i'm not sure about the original point of "operationVerb", but now it will be used to be mapped as api method name in other languages' code-generator. e.g. in java, we will use a method named`readNamespacedPod`, it is equal to client-go's `CoreV1().Pods(<ns>).Get(<n>)`. we have to keep consistent w/ built-ins otherwise the generated method from CRD will turn out to be sth different like `getNamespacedPod`. this can be confusing to downupstream users of published openapi docs..

Reply:
`operationVerb` is used to compose the operation ID, which ends up being in the openapi-generated method name

I don't know about the `x-kubernetes-action` extension. The existing code misses "list" and "deletecollection". Note that the extension [doesn't have "create" and "update"](https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/README.md#x-kubernetes-action), which is different from the authorization

HTTP method | x-kubernetes-action | operation ID | [auth request verb](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#determine-the-request-verb)
--- | --- | --- | --- 
GET | get | read | get
GET | list | list | list
PUT | put | replace | update
PATCH | patch | patch | patch
POST | post | create | create
DELETE | delete | delete | delete
DELETE | deletecollection | deletecollection | deletecollection

Reply:
great, i updated create/update to post/put